moduleroot/azure-pipelines/spec_job.yml.erb

```diff
@@ -0,0 +1,39 @@
+parameters:
+  name: ''
+  spec_command: ''
+  ruby_version: ''
+  puppet_gem_version: ''
+
+jobs:
+- job: ${{ parameters.name }}
+  variables:
+    puppet_gem_version: ${{ parameters.puppet_gem_version }}
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+  - task: UseRubyVersion@0
+    inputs:
+      versionSpec: ${{ parameters.ruby_version }}
+      addToPath: true # Optional
+  - script: |
+      bundle -v
+      rm -f Gemfile.lock
+      gem update --system $RUBYGEMS_VERSION --no-document
+    displayName: 'gem update'
+  - script: ruby -v | tee ruby_version.txt
+    displayName: 'save ruby_version file'
+  - task: 1ESLighthouseEng.PipelineArtifactCaching.RestoreAndSaveCacheV1.RestoreAndSaveCache@1
```

Comment:
Perhaps this whole task should be wrapped in an `if` ?

Reply:
this task only runs if the condition is met.  So it is basically wrapped in an if statement.

Reply:
Fair enough